### PR TITLE
Use `cudf::lists::distinct` in Java binding

### PR DESCRIPTION
**Pull request by ttnghia**
Java binding has `dropListDuplicates` and `dropListDuplicatesWithKeysValues` APIs to create lists column having distinct list elements. Previously they have been implemented by calling to `cudf::lists::drop_list_duplicates`, which relies on stably sorting the input lists column to extract unique list elements.

This PR makes the following changes:
 * Modifying the underlying implementation of `dropListDuplicates` to use `cudf::lists::distinct`, which has `O(n)` time complexity, and
 * Adding a new JNI-dedicated function `cudf::jni::lists_distinct_by_key` so `dropListDuplicatesWithKeysValues` will call it instead of `cudf::lists::drop_list_duplicates`. This function with keys-values pair input is only needed by Spark's `create_map` function so its implementation will be kept inside the JNI layer instead of in libcudf.

### CODE REVIEW

File: java/src/main/native/src/ColumnViewJni.cu [hunk 1]
```diff
@@ -72,4 +80,50 @@ std::unique_ptr<cudf::column> generate_list_offsets(cudf::column_view const &lis
...
+                                                    rmm::cuda_stream_view stream) {
+  if (input.is_empty()) {
+    return empty_like(input.parent());
+  }
+
+  auto const child = input.get_sliced_child(stream);
+
+  // Genereate labels for the input list elements.
+  auto labels = rmm::device_uvector<cudf::size_type>(child.size(), stream);
+  cudf::detail::label_segments(input.offsets_begin(), input.offsets_end(), labels.begin(),
```
[the remark this answers is not in the document]
Comment (mythrocks) [reply java/src/main/native/src/ColumnViewJni.cu:94]: Also, the code that constructs the group-labels column for keys: That is pretty sleek.

**Review by mythrocks — CHANGES_REQUESTED** [review]
Good work!

One minor documentation rephrasing requested. Plus, we've relaxed the requirement for non-sliced input columns; we should add a test to confirm.

File: java/src/main/native/src/ColumnViewJni.cpp [hunk 2]
```diff
@@ -476,59 +474,18 @@ JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnView_dropListDuplicatesWithKey
   try {
     cudf::jni::auto_set_device(env);
     auto const input_cv = reinterpret_cast<cudf::column_view const *>(keys_vals_handle);
-    CUDF_EXPECTS(input_cv->offset() == 0, "Input column has non-zero offset.");
```
[the remark this answers is not in the document]
Comment (ttnghia) [reply java/src/main/native/src/ColumnViewJni.cpp:479]: Got it. I just added a test for that in Java.

**Review by mythrocks — APPROVED** [review]
LGTM!

**Comment by ttnghia** [comment]
@gpucibot merge

**Pull request merged**